app/javascript/components/WorkflowsPage/WorkflowList.tsx

```diff
@@ -65,18 +66,20 @@ const WorkflowList = () => {
                   ) : (
                     <Button
                       color="danger"
-                      onClick={asyncVoid(async () => {
-                        try {
-                          setDeletingWorkflow({ ...deletingWorkflow, state: "deleting" });
-                          await deleteWorkflow(deletingWorkflow.id);
-                          setWorkflows(workflows.filter((workflow) => workflow.external_id !== deletingWorkflow.id));
-                          setDeletingWorkflow(null);
-                          showAlert("Workflow deleted!", "success");
-                        } catch (e) {
-                          assertResponseError(e);
-                          showAlert("Sorry, something went wrong. Please try again.", "error");
-                        }
-                      })}
+                      onClick={() => {
+                        setDeletingWorkflow({ ...deletingWorkflow, state: "deleting" });
+                        router.delete(Routes.workflow_path(deletingWorkflow.id), {
+                          onSuccess: () => {
+                            setWorkflows(workflows.filter((workflow) => workflow.external_id !== deletingWorkflow.id));
+                            setDeletingWorkflow(null);
+                            showAlert("Workflow deleted!", "success");
```

Comment:
You shouldn't have to do this anymore.
- You should have a `only: ["workflows"]` partial reload so it re-renders the correct list of workflow automatically.
- Flash message should be automatically set from the controller. See how we do it [in the admin layout](https://github.com/antiwork/gumroad/blob/main/app/javascript/layouts/Admin.tsx#L24).